### PR TITLE
Refactor provision.sh into cohesive services

### PR DESCRIPTION
**Pull request by jbarciauskas**
The goal here is to make a provision.sh that can be easily start up different common sets of configurations, e.g. everything or just individual services, or arbitrary combinations.

### CODE REVIEW

File: provision-ida.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+echo -e "${GREEN}Running migrations for ${name}...${NC}"
+docker exec -t edx.devstack.${name}  bash -c 'source /edx/app/$1/$1_env && cd /edx/app/$1/$1/ && make migrate' -- "$name"
+
+echo -e "${GREEN}Creating super-user for ${name}...${NC}"
+docker exec -t edx.devstack.${name}  bash -c 'source /edx/app/$1/$1_env && echo "from django.contrib.auth import get_user_model; User = get_user_model(); User.objects.create_superuser(\"edx\", \"edx@example.com\", \"edx\") if not User.objects.filter(username=\"edx\").exists() else None" | python /edx/app/$1/$1/manage.py shell' -- "$name"
+
+./provision-ida-user.sh $name $port
+
+# Compile static assets last since they are absolutely necessary for all services. This will allow developers to get
```
Comment (clintonb) [inline provision-ida.sh:17]: ...since they are **not** absolutely necessary...

File: provision-lms.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+docker exec -t edx.devstack.lms  bash -c 'source /edx/app/edxapp/edxapp_env && python /edx/app/edxapp/edx-platform/manage.py lms --settings=devstack_docker manage_user edx edx@example.com --superuser --staff'
+docker exec -t edx.devstack.lms  bash -c 'source /edx/app/edxapp/edxapp_env && echo "from django.contrib.auth import get_user_model; User = get_user_model(); user = User.objects.get(username=\"edx\"); user.set_password(\"edx\"); user.save()" | python /edx/app/edxapp/edx-platform/manage.py lms shell  --settings=devstack_docker'
+
+# Enable the LMS-E-Commerce integration
+docker exec -t edx.devstack.lms  bash -c 'source /edx/app/edxapp/edxapp_env && python /edx/app/edxapp/edx-platform/manage.py lms --settings=devstack_docker configure_commerce'
+
+# Create demo course and users
+docker exec -t edx.devstack.lms  bash -c '/edx/app/edx_ansible/venvs/edx_ansible/bin/ansible-playbook /edx/app/edx_ansible/edx_ansible/playbooks/edx-east/demo.yml -v -c local -i "127.0.0.1," --extra-vars="COMMON_EDXAPP_SETTINGS=devstack_docker"'
+
+docker restart edx.devstack.lms
```
Comment (clintonb) [inline provision-lms.sh]: Please add a comment explaining the need to restart.

File: provision.sh [hunk 3]
```diff
@@ -37,76 +38,15 @@ echo -e "${GREEN}Creating databases and users...${NC}"
...
 
-# Bring the rest of the services online
-docker-compose up -d
+./provision-lms.sh
 
-# Load database dumps for the largest databases to save time
-./load-db.sh ecommerce
-./load-db.sh edxapp
-./load-db.sh edxapp_csmh
+# TODO: No special actions needed for studio?
```
Comment (clintonb) [inline provision.sh]: There are no special actions as far as I know. LMS does it all.

File: provision.sh [hunk 4]
```diff
@@ -37,76 +38,15 @@ echo -e "${GREEN}Creating databases and users...${NC}"
...
-docker exec -t edx.devstack.ecommerce bash -c 'source /edx/app/ecommerce/ecommerce_env && python /edx/app/ecommerce/ecommerce/manage.py oscar_populate_countries' &
-
-# TODO Create discovery tenant with correct credentials (ECOM-6565)
-docker exec -t edx.devstack.discovery bash -c 'source /edx/app/discovery/discovery_env && python /edx/app/discovery/discovery/manage.py create_or_update_partner --code edx --name edX --courses-api-url "http://edx.devstack.lms:18000/api/courses/v1/" --ecommerce-api-url "http://edx.devstack.ecommerce:18130/api/v2/" --organizations-api-url "http://edx.devstack.lms:18000/api/organizations/v0/" --oidc-url-root "http://edx.devstack.lms:18000/oauth2" --oidc-key discovery-key --oidc-secret discovery-secret'
-docker exec -t edx.devstack.discovery bash -c 'source /edx/app/discovery/discovery_env && python /edx/app/discovery/discovery/manage.py refresh_course_metadata'
-docker exec -t edx.devstack.discovery bash -c 'source /edx/app/discovery/discovery_env && python /edx/app/discovery/discovery/manage.py update_index' &
+./provision-ecommerce.sh
+./provision-discovery.sh
+./provision-ida.sh credentials 18150
```
Comment (clintonb) [inline provision.sh]: Credentials needs the same love. We recently added a [create_or_update_site management command](https://github.com/edx/credentials/blob/master/credentials/apps/core/management/commands/create_or_update_site.py). I can add the call to `provision-credentials.sh`, if you make the file, once this is merged.

File: provision-discovery.sh [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Provisioning script for the discovery service
+
+./provision-ida.sh discovery 18381
+
+# TODO Create discovery tenant with correct credentials (ECOM-6565)
+docker exec -t edx.devstack.discovery bash -c 'source /edx/app/discovery/discovery_env && python /edx/app/discovery/discovery/manage.py create_or_update_partner --code edx --name edX --courses-api-url "http://edx.devstack.lms:18000/api/courses/v1/" --ecommerce-api-url "http://edx.devstack.ecommerce:18130/api/v2/" --organizations-api-url "http://edx.devstack.lms:18000/api/organizations/v0/" --oidc-url-root "http://edx.devstack.lms:18000/oauth2" --oidc-key discovery-key --oidc-secret discovery-secret'
+docker exec -t edx.devstack.discovery bash -c 'source /edx/app/discovery/discovery_env && python /edx/app/discovery/discovery/manage.py refresh_course_metadata'
+docker exec -t edx.devstack.discovery bash -c 'source /edx/app/discovery/discovery_env && python /edx/app/discovery/discovery/manage.py update_index'
+# TODO Create credentials tenant (ECOM-6566)
```
Comment (clintonb) [inline provision-discovery.sh:8]: Remove.

**Review by clintonb — APPROVED** [review]
A few minor issues but, otherwise, this is great!

File: provision-discovery.sh (hunk 5 above)
Comment (jbarciauskas) [reply provision-discovery.sh:8]: Should I just comment out provision-discovery.sh from provision.sh? Part of the goal here was to make it easy to specify exactly what services you want to run.

Comment (clintonb) [reply provision-discovery.sh:8]: Yes. Our primary goal is edxapp.

**Pull request merged**